osgp/shared/osgp-dto/src/main/java/org/opensmartgridplatform/dto/valueobjects/smartmetering/EventDto.java

```diff
@@ -31,4 +32,8 @@ public EventDto(final DateTime timestamp, final Integer eventCode, final Integer
         this.eventCounter = eventCounter;
         this.eventLogCategoryName = eventLogCategoryName;
     }
+
+    public void setEventTypeDto(EventTypeDto eventTypeDto) {
```

Comment:
I would move this to the constructor, so all elements are immutable.

Reply:
This is not possible, cause we add the EventType afterwards in the EventService, based on the eventCode